examples/Coffee_Machine_FSM/coffee.py

```diff
@@ -0,0 +1,201 @@
+from enum import Enum
+from typing import Union, List
+
+
+
+class StatusCode(Enum):
+    OK = 1
+    ERROR = 0
+    COFFEE_PRODUCED = 2
+
+
+
+class State:
+    def __init__(self, name: str, coffee_machine: "CoffeeMachine") -> None:
+        self.name: str = name
+        self.coffee_machine: CoffeeMachine = coffee_machine
+
+    def handle_input(self, input_alphabet: Union[str, List[str]]) -> int:
+        raise NotImplementedError("This method will be implemented by subclasses")
+
+    def handle_output(self, input_alphabet: Union[str, List[str]]) -> int:
+        raise NotImplementedError("This method will be implemented by subclasses")
+
+    def transition(self, input_alphabet: Union[str, List[str]]) -> int:
+        raise NotImplementedError("This method will be implemented by subclasses")
+
+
+
+class CoffeeMachine:
+    def __init__(self) -> None:
+        self.state_a = StateA('a', self)  
```

Comment:
Remove these assignments. Rather put the names as inherent class variables of the states